exercises/wordy/example.rs

```diff
@@ -18,74 +12,57 @@ impl Token {
     }
 
     fn is_operator(&self) -> bool {
-        self.value == String::from("plus")
-            || self.value == String::from("minus")
-            || self.value == String::from("multiplied")
-            || self.value == String::from("divided")
+        self.value == "plus"
+            || self.value == "minus"
+            || self.value == "multiplied"
+            || self.value == "divided"
     }
 }
 
-pub fn answer(c: &str) -> Option<isize> {
-    WordProblem::new(c).answer()
-}
+pub fn answer(c: &str) -> Option<i32> {
+    let mut t = tokens(c);
+    let mut result: i32 = 0;
+    let mut opr = "plus";
 
-impl WordProblem {
-    fn new(c: &str) -> Self {
-        WordProblem {
-            command: String::from(c),
+    if t.len() <= 1 {
+        None
+    } else {
+        while t.len() > 1 {
+            result = evaluate(result, opr, operand(&t.remove(0)));
+            opr = operator(&t.remove(0));
         }
+        result = evaluate(result, opr, operand(&t.remove(0)));
+        Some(result)
     }
+}
 
-    fn answer(&self) -> Option<isize> {
-        let mut t = self.tokens();
-        let mut result: isize = 0;
-        let mut opr = "plus".to_string();
-
-        if t.len() <= 1 {
-            None
-        } else {
-            while t.len() > 1 {
-                result = self.evaluate(result, opr, self.operand(&t.remove(0)));
-                opr = self.operator(&t.remove(0));
-            }
-            result = self.evaluate(result, opr, self.operand(&t.remove(0)));
-            Some(result)
-        }
-    }
-
-    fn evaluate(&self, mut r: isize, operator: String, operand: isize) -> isize {
-        if operator == String::from("plus") {
-            r += operand;
-            r
-        } else if operator == String::from("minus") {
-            r -= operand;
-            r
-        } else if operator == String::from("multiplied") {
-            r *= operand;
-            r
-        } else if operator == String::from("divided") {
-            r /= operand;
-            r
-        } else {
-            r
-        }
+fn evaluate(mut r: i32, operator: &str, operand: i32) -> i32 {
+    if operator == "plus" {
+        r += operand;
+    } else if operator == "minus" {
+        r -= operand;
+    } else if operator == "multiplied" {
+        r *= operand;
+    } else if operator == "divided" {
+        r /= operand;
     }
+    r
+}
 
-    fn operand(&self, t: &Token) -> isize {
-        t.value.parse().unwrap()
-    }
+fn operand(t: &Token) -> i32 {
+    t.value.parse().unwrap()
```

Comment:
The `unwrap` here is also surprising: `answer("plus 1 2")` will panic, instead of returning `None`. This is another potential improvement we could make to this exercise, though it might not fit into this PR.

(In an ideal world, it might return `Err(wordy::Error::ReversePolishNotation)`, but that's probably not a change we actually want to make to this exercise.)

Reply:
One might also imagine testing that `"1 plus plus 2"` results in `None`. While usually I'm not so interested, I do have a suspicion that without such cases, it'd be possible to get away with parsers that are a bit too fragile. The parser in this example is indeed a little too permissive.